datafusion/physical-plan/src/memory.rs

```diff
@@ -139,13 +140,18 @@ pub trait LazyBatchGenerator: Send + Sync + fmt::Debug + fmt::Display {
 ///
 /// This plan generates output batches lazily, it doesn't have to buffer all batches
 /// in memory up front (compared to `MemorySourceConfig`), thus consuming constant memory.
+/// We now add a `cooperative` flag to
+/// let it optionally yield back to the runtime periodically.
+/// Default is `true`, meaning it will yield back to the runtime for cooperative scheduling.
 pub struct LazyMemoryExec {
     /// Schema representing the data
     schema: SchemaRef,
     /// Functions to generate batches for each partition
     batch_generators: Vec<Arc<RwLock<dyn LazyBatchGenerator>>>,
     /// Plan properties cache storing equivalence properties, partitioning, and execution mode
     cache: PlanProperties,
+    /// Indicates whether to enable cooperative yielding mode.
+    cooperative: bool,
```

Comment:
Also enable LazyMemoryExec as a built-in, because it's a normal case, for example, the following cases will hit this:

```rust
SET datafusion.execution.target_partitions = 1;

SELECT SUM(value) FROM range(1, 50000000000);
```